core/block.js

```diff
@@ -1024,7 +1024,7 @@ Blockly.Block.prototype.setOnChange = function(onchangeFn) {
 Blockly.Block.prototype.getField = function(name) {
   for (var i = 0, input; (input = this.inputList[i]); i++) {
     for (var j = 0, field; (field = input.fieldRow[j]); j++) {
-      if (field.name == name) {
+      if (typeof field.name !== "undefined" && field.name == name) {
```

Comment:
I was wondering why `==` was used here instead of `===`.  My thinking with using `typeof field.name !== "undefined"` was that it would sort of indicate that since the type of a field name must be `string` or `undefined`, we're filtering the list on names with a `string` type.  But if we're requiring the developer to use string field names, and I'm able to change `==` to `===`, that's definitely clearer!